src/crimson/osd/backfill_state.cc

```diff
@@ -409,7 +409,35 @@ BackfillState::PrimaryScanning::react(PrimaryScanned evt)
   LOG_PREFIX(BackfillState::PrimaryScanning::react::PrimaryScanned);
   DEBUGDPP("", pg());
   backfill_state().backfill_info = std::move(evt.result);
-  return transit<Enqueuing>();
+  if (!backfill_state().next) {
```

Comment:
We are not checking here the actual value of `*backfill_state().next`, we just check if it has a value.